code/src/main/java/com/onlinecontacttracing/messaging/EmailSender.java

```diff
@@ -0,0 +1,116 @@
+package com.onlinecontacttracing.messaging;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+import com.google.api.services.gmail.model.Label;
+import com.google.api.services.gmail.model.ListLabelsResponse;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.mail.internet.MimeMessage;
+import javax.mail.MessagingException;
+import java.util.logging.Logger;
+import com.onlinecontacttracing.storage.PersonEmail;
+import com.onlinecontacttracing.storage.NotificationBatch;
+import org.apache.commons.codec.binary.Base64;
+import com.google.api.services.gmail.model.Message;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.io.ByteArrayOutputStream;
+import java.util.Properties;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
+/**
+* Email sender for sending emails to user's contacts.
```

Comment:
plz include more detail - who will get the emails, who will they be sent by, etc.